infertrade/utilities/operations.py

```diff
@@ -402,8 +402,23 @@ def limit_allocation(dataframe: pd.DataFrame, allocation_lower_limit: Union[int,
             ] = allocation_lower_limit
     return dataframe
 
+def daily_stop_loss(dataframe: pd.DataFrame, loss_limit: float) -> pd.DataFrame:
+    """
+    This function calculate daily percent change and restricts allocation if percent change in price is less than -loss_limit
+    
+    params:
+    allocated_dataframe
+    loss_limit: percent_change below which allocation is 0
+    """
+    # TODO improve this function to calculate stop loss
+    dataframe["allocation"][dataframe.price.pct_change()<-loss_limit]=0
```

Comment:
The loss is price change * position size taken - we need to multiple by the allocation value of the previous day before comparing versus the permitted loss limit.